src/Features/Core/Portable/Navigation/NavigableItemFactory.DeclaredSymbolNavigableItem.cs

```diff
@@ -2,77 +2,94 @@
 
 using System;
 using System.Collections.Immutable;
-using System.Threading;
-using Microsoft.CodeAnalysis.ErrorReporting;
 using Microsoft.CodeAnalysis.FindSymbols;
-using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Tags;
 using Microsoft.CodeAnalysis.Text;
-using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Navigation
 {
     internal partial class NavigableItemFactory
     {
         internal class DeclaredSymbolNavigableItem : INavigableItem
         {
-            public ImmutableArray<TaggedText> DisplayTaggedParts => _lazyDisplayTaggedParts.Value; 
+            private readonly DeclaredSymbolInfo _declaredSymbolInfo;
 
             public Document Document { get; }
-            public Glyph Glyph => this.SymbolOpt?.GetGlyph() ?? Glyph.Error;
+
+            public ImmutableArray<TaggedText> DisplayTaggedParts
+                => ImmutableArray.Create(new TaggedText(
+                    TextTags.Text, _declaredSymbolInfo.Name + _declaredSymbolInfo.NameSuffix));
+
+            public Glyph Glyph => GetGlyph(_declaredSymbolInfo.Kind, _declaredSymbolInfo.Accessibility);
```

Comment:
Glyph is not computed entirely from data in the index.  No going back to a symbol.